supervisor/addons/addon.py

```diff
@@ -1399,3 +1430,9 @@ async def watchdog_container(self, event: DockerContainerStateEvent) -> None:
             ContainerState.UNHEALTHY,
         ]:
             await self._restart_after_problem(event.state)
+
+    def refresh_cache(self) -> Awaitable[None]:
```

Comment:
Do we need this method still? Afaik, we can directly rely on `super().refresh_cache()` here now, as we update the store on store update :thinking:

Reply:
We don't need to do this its true. I just didn't like having a public method in the class that couldn't be called. Because if someone called it then it would refresh the cache in `Addon` not `StoreAddon` which is wrong for like 99% of addons

Reply:
Hm, I see your point.

I guess it is a bit confusing that `load()` now checks for `self.is_detached`, and here we do it again. But it makes sense since cache refreshing for add-ons from store is taken care from store side.

Maybe we should name the two functions to make it less confusing? :thinking: 

Where is `self.addon_store.refresh_cache()` actually implemented? :thinking:

Reply:
> I guess it is a bit confusing that load() now checks for self.is_detached, and here we do it again. But it makes sense since cache refreshing for add-ons from store is taken care from store side.

@agners The reason for this is because although it would fine if I removed the check from `load`, it would cause the cache in the corresponding store addon to refresh twice. Once on load of `StoreManager`, once on load of the installed `Addon`. The alternative is filtering out installed addons in `StoreManager` when doing that refresh but this seemed cleaner.

> Where is self.addon_store.refresh_cache() actually implemented? 🤔

In `AddonModel`. It's shared between `StoreAddon` and `Addon` because installed addons may use it as well, if they are detached.

> Maybe we should name the two functions to make it less confusing? 🤔

I don't think this is a good idea. There's a lot of places in supervisor where we get a list of `AnyAddon`, which could be a `StoreAddon` and could be an `Addon`. It's generally better to keep the interface the same when possible IMO.

Reply:
> @agners The reason for this is because although it would fine if I removed the check from load, it would cause the cache in the corresponding store addon to refresh twice. Once on load of StoreManager, once on load of the installed Addon. 

Yes after following code paths, I do see that. But it isn't immediately apparent from reading the code. Which is kinda my reason of suggesting a rename: It makes reading the code a easier.

Looking at the code again, I think it make sense to use `super().self.refresh_path_cache()` in `load()`, this way it is congruent with the `with_ *` properties. In the end it is all minor details. It just is when stumbling as a reviewer, whoever is touching the code next, will probably get confused as well. And potentially introduce a bug because he tries to "optimize" and removes the if in `load()` :see_no_evil:

Reply:
I'll just block it from addon's interface. We can call the superclass one from load in the edge case and the one in addon will be marked as non-functional so the interface is intelligble

Reply:
I left the method as it, its just not used anywhere. There's not really a good pythonic way to block a method at a descendant I can see and it seemed like the same amount of code to just have it implemented in case.